src/adapters/ws/client.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * WS client adapter
+ */
+import Adapter from '../../lib/adapter.js';
+import GleeMessage from '../../lib/message.js';
+import ws from 'ws';
+
+
+class WsClientAdapter extends Adapter {
+    private client: ws
+
+    name(): string {
+        return 'WS adapter';
+    }
+
+    async connect(): Promise<this> {
+        return this._connect();
+    }
+
+    async send(message: GleeMessage) {
+        return this._send(message);
+    }
+
+    _connect(): Promise<this> {
+        return new Promise((resolve) => {
+
+            const subscribedChannels = this.getSubscribedChannels();
```

Comment:
This is not being used so it means you're not using the channels information at all. Check my comment below 👇